core/src/main/scala/org/apache/spark/deploy/rest/RestSubmissionClient.scala

```diff
@@ -58,7 +60,11 @@ import org.apache.spark.{Logging, SPARK_VERSION => sparkVersion, SparkConf}
 private[spark] class RestSubmissionClient(master: String) extends Logging {
   import RestSubmissionClient._
 
-  private val supportedMasterPrefixes = Seq("spark://", "mesos://")
+  private val supportedMasterPrefixes = Map(
+    "spark://" -> "http://",
+    "mesos://" -> "http://",
+    "mesoss://" -> "https://"
```

Comment:
This also warrents some documentation I think. Can you put it in running-on-mesos?